packages/workato-adapter/src/rlm.ts

```diff
@@ -0,0 +1,270 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { EOL } from 'os'
+import _ from 'lodash'
+import * as path from 'path'
+import JSZip from 'jszip'
+import { logger } from '@salto-io/logging'
+import { Change, DeployResult, ElemID, getChangeData, Element, InstanceElement, isInstanceElement, ReadOnlyElementsSource, Values } from '@salto-io/adapter-api'
+import { createSchemeGuard, GetLookupNameFunc, getParent, resolveValues, ResolveValuesFunc, safeJsonStringify } from '@salto-io/adapter-utils'
+import { retry } from '@salto-io/lowerdash'
+import Joi from 'joi'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import WorkatoClient from './client/client'
+import { CONNECTION_TYPE, RECIPE_CODE_TYPE, RECIPE_TYPE } from './constants'
+import { isTypeChange } from './utils'
+
+const { withRetry } = retry
+const { intervals } = retry.retryStrategies
+const log = logger(module)
+
+const MAX_RETRIES = 60
+const INTERVAL_TIME = 2000
+const RESTART_RECIPES_QUERY_PARAMS:Record<string, string> = { restart_recipes: 'true' }
+const IMPORT_HEADER = { 'Content-Type': 'application/octet-type' }
+
+type JsonRecipeConfigAccountID = {
+  zip_name: string
+  name: string
+  folder: string
+}
+
+type JsonRecipeConfig = {
+  keyword: string
+  provider: string
+  skip_validation: boolean
+  account_id: JsonRecipeConfigAccountID | null
+}
+
+type JsonRecipe = {
+  name: string
+  description: string
+  version: number
+  private?: boolean
+  concurrency?: number
+  code: Values
+  config: JsonRecipeConfig[]
+}
+
+type folderElement = {
```

Comment:
nits:
1.  
```suggestion
type FolderElement = {
```
2. why `Element` (here and in other type names)? this kind of a reserved term inside salto that usually has a different meaning